src/ngx_http_lua_upstream_module.c

```diff
@@ -141,6 +148,92 @@ ngx_http_lua_upstream_get_upstreams(lua_State * L)
     return 1;
 }
 
+static int
+ngx_http_lua_upstream_add_upstream_peer(lua_State * L)
+{
+    ngx_str_t                    host;
+    ngx_http_upstream_server_t   *us;
+    ngx_http_upstream_srv_conf_t *uscf;
+    ngx_url_t                    u;
+    ngx_http_request_t           *r;
+    ngx_int_t                    weight, max_fails;
+    ngx_uint_t                   backup;
+    time_t                       fail_timeout;
+    u_char                       *p;
+
+    if (lua_gettop(L) != 6) {
+        /*
+        * "upstream name", "host:port", "weight", "max_fails", "fail_timeout", "backup"
+        */
+        return luaL_error(L, "exactly 6 arguments expected");
+    }
+
+    r = ngx_http_lua_get_request(L);
+    if (r == NULL) {
+        lua_pushnil(L);
+        lua_pushliteral(L, "get request error \n");
+        return 2;
+    }
+
+    host.data = (u_char *) luaL_checklstring(L, 1, &host.len);
+
+    ngx_memzero(&u, sizeof(ngx_url_t));
```

Comment:
You can declare `u` like so: `ngx_url_t                    u = { 0 };` to automatically zero it.